src/convertToObject.js

```diff
@@ -3,16 +3,26 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an exampl
+ * in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';')
+    .map(property => property.trim().split(':'))
+    .filter(property => property.length === 2)
+    .map(([property, value]) => [property.trim(), value.trim()])
+    .reduce((cssObject, [key, value]) => {
+      cssObject[key] = value;
+
```

Comment:
You have 4 iterations here, while we can make it in 1 iteration. Let's optimize our algorithm